const array when borrow needs '&'

https://github.com/cloud-hypervisor/cloud-hypervisor/blob/46953db3ca206f42ad981ba524bee2427ba7adbc/vmm/src/vm.rs#L1596

the correct:
```bash
...
        for sig in &HANDLED_SIGNALS {
            unblock_signal(*sig).unwrap();
        }
...
```

the same correcting method does at     https://github.com/cloud-hypervisor/cloud-hypervisor/blob/46953db3ca206f42ad981ba524bee2427ba7adbc/src/main.rs#L516
```bash
...
    for sig in &vmm::vm::HANDLED_SIGNALS {
        if let Err(e) = block_signal(*sig) {
...
```

hi -

i'm ignorant of the details of rust, but i was unable to build due to this issue with HANDLED_SIGNALS:

```
error[E0277]: `[i32; 3]` is not an iterator
    --> vmm/src/vm.rs:1657:20
     |
1657 |         for sig in HANDLED_SIGNALS {
     |                    ^^^^^^^^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
     |
     = help: the trait `Iterator` is not implemented for `[i32; 3]`
     = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`
     = note: required because of the requirements on the impl of `IntoIterator` for `[i32; 3]`
     = note: required by `into_iter
```

if i hack in the &, i get another issue later on:

```
   --> src/main.rs:516:16
    |
516 |     for sig in vmm::vm::HANDLED_SIGNALS {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^ borrow the array with `&` or call `.iter()` on it to iterate over it
    |
```
rustc 1.50.0
at commit b519975fc859 ("build: Use Ubuntu impish (21.10) for workers")

--- a/src/main.rs
+++ b/src/main.rs
@@ -513,8 +513,8 @@ fn start_vmm(cmd_arguments: ArgMatches) -> Result<Option<String>, Error> {
     // Before we start any threads, mask the signals we'll be
     // installing handlers for, to make sure they only ever run on the
     // dedicated signal handling thread we'll start in a bit.
-    for sig in vmm::vm::HANDLED_SIGNALS {
-        if let Err(e) = block_signal(sig) {
+    for sig in &vmm::vm::HANDLED_SIGNALS {
+        if let Err(e) = block_signal(*sig) {
             eprintln!("Error blocking signals: {}", e);
         }
     }

--- a/vmm/src/vm.rs
+++ b/vmm/src/vm.rs
@@ -1654,8 +1654,8 @@ impl Vm {
         on_tty: bool,
         exit_evt: &EventFd,
     ) {
-        for sig in HANDLED_SIGNALS {
-            unblock_signal(sig).unwrap();
+        for sig in &HANDLED_SIGNALS {
+            unblock_signal(*sig).unwrap();
         }
 
         for signal in signals.forever() {

